src/D2L.Bmx/Aws/AwsCredsCache.cs

```diff
@@ -0,0 +1,65 @@
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+namespace D2L.Bmx.Aws;
+
+internal class AwsCredsCache() {
+	private static readonly JsonSerializerOptions _options =
+		new() { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull };
+
+	public void SaveToFile( string Org, string User, AwsRole role, AwsCredentials credentials ) {
+		if( !Directory.Exists( BmxPaths.BMX_DIR ) ) {
+			return;
+		}
+		if( Org == "" || User == "" ) { //It's not set in config file
+			return;
+		}
+		var newEntry = new AwsCacheModel(
+			Org,
+			User,
+			role.RoleArn,
+			credentials );
+		List<AwsCacheModel> allEntries = GetAllCache();
+		allEntries.RemoveAll( o => o.Credentials.Expiration < DateTime.Now );
+		allEntries.Add( newEntry );
+		string jsonString = JsonSerializer.Serialize( allEntries, SourceGenerationContext.Default.ListAwsCacheModel );
+
+		WriteTextToFile( BmxPaths.AWS_CREDS_CACHE_FILE_NAME, jsonString );
+	}
+	private static void WriteTextToFile( string path, string content ) {
+		var op = new FileStreamOptions {
+			Mode = FileMode.Create,
+			Access = FileAccess.Write,
+		};
+		if( !RuntimeInformation.IsOSPlatform( OSPlatform.Windows ) ) {
+			op.UnixCreateMode = UnixFileMode.UserRead | UnixFileMode.UserWrite;
+		}
+		using var writer = new StreamWriter( path, op );
+		writer.Write( content );
+	}
+
+	public List<AwsCacheModel> GetAllCache() {
+		string CacheFileName = BmxPaths.AWS_CREDS_CACHE_FILE_NAME;
+		if( !File.Exists( CacheFileName ) ) {
+			return new();
+		}
+		try {
+			string sessionsJson = File.ReadAllText( BmxPaths.AWS_CREDS_CACHE_FILE_NAME );
+			return JsonSerializer.Deserialize( sessionsJson, SourceGenerationContext.Default.ListAwsCacheModel )
+				?? new();
+		} catch( JsonException ) {
+			return new();
+		}
+	}
+
+	public AwsCredentials? GetCachedSession( string Org, string User, AwsRole role, int Cache ) {
+		// Main config is at ~/.bmx/config
```

Comment:
What's this here for?

Reply:
Get a specified session where Org, User, and role all matches, and expire time is greater than now + Cache

Reply:
The comment I mean sorry 
> // Main config is at ~/.bmx/config

Reply:
Ah okay yea can we remove it then if we're not using that config file so no need to reference this
```suggestion
```